kernel/src/table_properties/deserialize.rs

```diff
@@ -58,81 +56,88 @@ fn try_parse(props: &mut TableProperties, k: &str, v: &str) -> Option<()> {
-        "delta.logRetentionDuration" => props.log_retention_duration = Some(parse_interval(v)?),
-        "delta.enableExpiredLogCleanup" => props.enable_expired_log_cleanup = Some(parse_bool(v)?),
-        "delta.randomizeFilePrefixes" => props.randomize_file_prefixes = Some(parse_bool(v)?),
-        "delta.randomPrefixLength" => props.random_prefix_length = Some(parse_positive_int(v)?),
+        "delta.logRetentionDuration" => props.log_retention_duration = parse_interval(v)?,
+        "delta.enableExpiredLogCleanup" => props.enable_expired_log_cleanup = parse_bool(v)?,
+        "delta.randomizeFilePrefixes" => props.randomize_file_prefixes = parse_bool(v)?,
+        "delta.randomPrefixLength" => props.random_prefix_length = parse_positive_int(v)?,
         "delta.setTransactionRetentionDuration" => {
-            props.set_transaction_retention_duration = Some(parse_interval(v)?)
-        }
-        "delta.targetFileSize" => props.target_file_size = Some(parse_positive_int(v)?),
-        "delta.tuneFileSizesForRewrites" => {
-            props.tune_file_sizes_for_rewrites = Some(parse_bool(v)?)
+            props.set_transaction_retention_duration = parse_interval(v)?
         }
+        "delta.targetFileSize" => props.target_file_size = parse_positive_int(v)?,
+        "delta.tuneFileSizesForRewrites" => props.tune_file_sizes_for_rewrites = parse_bool(v)?,
         "delta.checkpointPolicy" => props.checkpoint_policy = CheckpointPolicy::try_from(v).ok(),
-        "delta.enableRowTracking" => props.enable_row_tracking = Some(parse_bool(v)?),
-        "delta.enableInCommitTimestamps" => {
-            props.enable_in_commit_timestamps = Some(parse_bool(v)?)
-        }
+        "delta.enableRowTracking" => props.enable_row_tracking = parse_bool(v)?,
+        "delta.enableInCommitTimestamps" => props.enable_in_commit_timestamps = parse_bool(v)?,
         "delta.inCommitTimestampEnablementVersion" => {
-            props.in_commit_timestamp_enablement_version = Some(parse_non_negative(v)?)
+            props.in_commit_timestamp_enablement_version = parse_non_negative(v)?
         }
         "delta.inCommitTimestampEnablementTimestamp" => {
-            props.in_commit_timestamp_enablement_timestamp = Some(parse_non_negative(v)?)
+            props.in_commit_timestamp_enablement_timestamp = parse_non_negative(v)?
         }
-        _ => return None,
+        _ => return Err(Error::Generic("foo".to_string())),
     }
-    Some(())
+    Ok(())
 }
 
 /// Deserialize a string representing a positive (> 0) integer into an `Option<u64>`. Returns `Some`
 /// if successfully parses, and `None` otherwise.
-pub(crate) fn parse_positive_int(s: &str) -> Option<NonZero<u64>> {
+pub(crate) fn parse_positive_int(s: &str) -> DeltaResult<Option<NonZero<u64>>> {
     // parse as non-negative and verify the result is non-zero
-    NonZero::new(parse_non_negative(s)?)
+    parse_non_negative(s).map(|maybe_num| maybe_num.and_then(NonZero::<u64>::new))
```

Comment:
From the comment above I think we can remove these functions entirely. Also we can remove the optional inside it entirely, the result should capture what goes wrong with the parsing.